models/demos/llama3/tt/llama_rope.py

```diff
@@ -163,6 +169,11 @@ def get_rot_mats(self, position_idxs, return_rot_idxs=False):
             cos = cos[:, : self.batch_size, :, :]
             sin = sin[:, : self.batch_size, :, :]
 
+        if self.num_devices == 32 and self.batch_size > 1:
+            # On TG slice to num_batches_per_group
+            cos = cos[:, :8, :, :]
```

Comment:
Actually we don't need to slice for TG which has separate batch_per_device
We can just initialize `TtLlamaRotarySetup` with `batch_size = batch_per_device`. This setup is only used inside attention.
This is how I have done to integrate fused ROPE where I just 2x batch_size to initialize for Q and K together.